iRacing/iRacing.cs

```diff
@@ -666,108 +681,36 @@ public Data(DahlDesign dahlDesign)
-            Base.AddProp("Lap05FuelTargetDelta", 0);
-
-            Base.AddProp("Lap06Sector1Time", new TimeSpan(0));
-            Base.AddProp("Lap06Sector2Time", new TimeSpan(0));
-            Base.AddProp("Lap06Sector3Time", new TimeSpan(0));
-            Base.AddProp("Lap06Sector1Delta", 0);
-            Base.AddProp("Lap06Sector2Delta", 0);
-            Base.AddProp("Lap06Sector3Delta", 0);
-            Base.AddProp("Lap06Sector1Status", 0);
-            Base.AddProp("Lap06Sector2Status", 0);
-            Base.AddProp("Lap06Sector3Status", 0);
-            Base.AddProp("Lap06FuelTargetDelta", 0);
-
-            Base.AddProp("Lap07Sector1Time", new TimeSpan(0));
-            Base.AddProp("Lap07Sector2Time", new TimeSpan(0));
-            Base.AddProp("Lap07Sector3Time", new TimeSpan(0));
-            Base.AddProp("Lap07Sector1Delta", 0);
-            Base.AddProp("Lap07Sector2Delta", 0);
-            Base.AddProp("Lap07Sector3Delta", 0);
-            Base.AddProp("Lap07Sector1Status", 0);
-            Base.AddProp("Lap07Sector2Status", 0);
-            Base.AddProp("Lap07Sector3Status", 0);
-            Base.AddProp("Lap07FuelTargetDelta", 0);
-
-            Base.AddProp("Lap08Sector1Time", new TimeSpan(0));
-            Base.AddProp("Lap08Sector2Time", new TimeSpan(0));
-            Base.AddProp("Lap08Sector3Time", new TimeSpan(0));
-            Base.AddProp("Lap08Sector1Delta", 0);
-            Base.AddProp("Lap08Sector2Delta", 0);
-            Base.AddProp("Lap08Sector3Delta", 0);
-            Base.AddProp("Lap08Sector1Status", 0);
-            Base.AddProp("Lap08Sector2Status", 0);
-            Base.AddProp("Lap08Sector3Status", 0);
-            Base.AddProp("Lap08FuelTargetDelta", 0);
-
-            Base.AddProp("LapRecord", new TimeSpan(0));
+            Base.AttachDelegate("OptimalLapTime", () => optimalLapTime);
+
+            for (int i = 0; i < 7; i++)
+            {
+                string propIndex = $"{i + 1:00}";
+
+                Base.AttachDelegate($"Lap{propIndex}Sector1Time", () => TimeSpan.FromSeconds(sector1TimeList[i]));
+                Base.AttachDelegate($"Lap{propIndex}Sector2Time", () => TimeSpan.FromSeconds(sector2TimeList[i]));
+                Base.AttachDelegate($"Lap{propIndex}Sector3Time", () => TimeSpan.FromSeconds(sector3TimeList[i]));
+                Base.AddProp($"Lap{propIndex}Sector1Delta", 0);
+                Base.AddProp($"Lap{propIndex}Sector2Delta", 0);
+                Base.AddProp($"Lap{propIndex}Sector3Delta", 0);
+                Base.AttachDelegate($"Lap{propIndex}Sector1Status", () => sector1StatusList[i]);
+                Base.AttachDelegate($"Lap{propIndex}Sector2Status", () => sector2StatusList[i]);
+                Base.AttachDelegate($"Lap{propIndex}Sector3Status", () => sector3StatusList[i]);
+                Base.AttachDelegate($"Lap{propIndex}FuelTargetDelta", () => fuelTargetDeltas[i]);
+            }
+
+            Base.AttachDelegate("LapRecord", () => lapRecord);
```

Comment:
Can't decifer it completely from the changes, but lap recod delta is broken.

Reply:
lapRecord is being updated by LapRecords class, using a reference. Maybe this doesn't work with delegates. So I created a dedicated variable 'proplapRecords' to assign

Reply:
Now I have seen the issue, the properties Lap<nn>Sector<n>Delta are not updated correctly :-(

Reply:
Ok, maybe leave the lap timing system as it was?

Reply:
![image](https://user-images.githubusercontent.com/40788634/205490740-606bf385-41d6-4399-a644-570a5d55d059.png)
 Hmm, testing now it seems like all is in order.

Reply:
Also now check the lap records. Modifed the CSV file and rerun a session. The lap was dedected as new lap record and the CSV was update.

![image](https://user-images.githubusercontent.com/14548927/205491445-a9d2fd01-9eac-4d7f-9cf2-8298aa502fcb.png)